app/models/upload_form.rb

```diff
@@ -4,26 +4,66 @@ class UploadForm
 
   validate :validate_reading_csv
   validate :validate_csv
+  validate :validate_data_present
+  validate :validate_header
+
+  ADDRESS = BulkUpload::Address::ADDRESS
+  HEADER_COLS = BulkUpload::Address.headers
 
   def validate_reading_csv
-    @data ||= CSV.read(upload_file, col_sep: "\t")
+    result = CSV.read(upload_file, headers: true)
+    @data ||= result.to_a
+    @headers ||= result.headers.slice(0..HEADER_COLS.length - 1)
   rescue TypeError
     errors.add(:upload_file, :csv_error, message: "Choose a file before uploading")
   rescue CSV::MalformedCSVError, CSV::Parser::InvalidEncoding
-    errors.add(:upload_file, :csv_error, message: "Invalid file type")
+    errors.add(:upload_file, :csv_error, message: "File must be csv format")
   end
 
   def validate_csv
     errors.add(:upload_file, :csv_error, message: "Invalid data structure") unless csv_valid?
   end
 
+  def validate_data_present
+    errors.add(:upload_file, :csv_error, message: "File has no data") if first_row_empty?
+  end
+
+  def validate_header
```

Comment:
This validation is different from the one in line 23 and 27 - they both use a private function to determine if the form is valid. Maybe we can make them look the same?